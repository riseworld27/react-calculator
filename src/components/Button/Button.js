import PropTypes from 'prop-types';

import styles from './Button.module.css';

const Button = ({
  name, handleClick, color, backgroundColor, wide,
}) => (
  <button
    type="button"
    className={styles.button}
    onClick={handleClick}
    style={{ color, backgroundColor, width: wide ? '50%' : '25%' }}
  >
    {name}
  </button>
);

Button.propTypes = {
  name: PropTypes.string,
  handleClick: PropTypes.func,
  color: PropTypes.string,
  backgroundColor: PropTypes.string,
  wide: PropTypes.bool,
};

Button.defaultProps = {
  name: 'Empty',
  handleClick: () => {},
  color: 'white',
  backgroundColor: '#f49040',
  wide: false,
};

export default Button;
