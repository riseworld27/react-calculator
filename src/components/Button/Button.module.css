.button {
  height: 120px;
  text-align: center;
  border: 1px solid #d0d0d0;
  border-radius: 2.5px;
  font-size: 35px;
  background-color: #dfdfdf;
}
