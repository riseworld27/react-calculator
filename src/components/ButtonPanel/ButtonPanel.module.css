.buttonPanel {
  display: flex;
  flex-wrap: wrap;
  line-height: 100px;
  background-color: #d0d0d0;
}
