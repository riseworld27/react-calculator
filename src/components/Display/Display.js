import React from 'react';
import PropTypes from 'prop-types';

import styles from './Display.module.css';

const Display = ({ result }) => (
  <div className={styles.displayContainer}>
    <p className={styles.displayOutput}>{result}</p>
  </div>
);

Display.propTypes = {
  result: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

Display.defaultProps = {
  result: 0,
};

export default Display;
