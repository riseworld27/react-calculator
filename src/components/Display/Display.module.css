.displayContainer {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  background-color: #202020;
  height: 100px;
  color: white;
  text-align: right;
  padding-right: 10px;
}

.displayOutput {
  font-size: 50px;
}
