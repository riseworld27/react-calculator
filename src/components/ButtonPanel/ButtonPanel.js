import PropTypes from 'prop-types';

import Button from '../Button/Button';
import styles from './ButtonPanel.module.css';

const buttonList = {
  group1: ['AC', '+/-', '%', '/'],
  group2: ['7', '8', '9', 'x'],
  group3: ['4', '5', '6', '-'],
  group4: ['1', '2', '3', '+'],
  group5: ['0', '.', '='],
};

const opList = ['+', '-', 'x', '/', '='];

const Panel = ({ clickHandler }) => (
  <div className={styles.buttonPanel}>
    {
      Object.keys(buttonList).map(key => (
        buttonList[key].map(value => (
          <Button
            key={value}
            name={value}
            group={key}
            handleClick={clickHandler}
            backgroundColor={opList.includes(value) ? undefined : '#dfdfdf'}
            color={opList.includes(value) ? 'white' : 'black'}
            wide={['0'].includes(value) ? true : undefined}
          />
        ))))
    }
  </div>
);

Panel.propTypes = {
  clickHandler: PropTypes.func,
};

Panel.defaultProps = {
  clickHandler: () => {},
};

export default Panel;
