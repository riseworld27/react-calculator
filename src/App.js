import React, { useState, useEffect } from 'react';

import Display from './components/Display/Display';
import ButtonPanel from './components/ButtonPanel/ButtonPanel';
import calculate from './utils/calculate';
import styles from './App.module.css';

const App = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);
  const [display, setDisplay] = useState(null);

  const handleClick = buttonName => {
    const result = calculate({ total, next, operation }, buttonName);
    setTotal(result.total);
    setNext(result.next);
    setOperation(result.operation);
  };

  useEffect(() => {
    if (operation === null) {
      if (total === null) {
        setDisplay(<Display result="0" />);
      } else {
        setDisplay(<Display result={total} />);
      }
    } else if (next === null) {
      setDisplay(<Display result={`${total} ${operation}`} />);
    } else {
      setDisplay(<Display result={`${total} ${operation} ${next}`} />);
    }
  }, [total, next, operation]);

  return (
    <div className={styles.container}>
      {display}
      <ButtonPanel clickHandler={e => handleClick(e.target.innerText)} />
    </div>
  );
};

export default App;
