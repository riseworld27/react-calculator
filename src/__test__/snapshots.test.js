import renderer from 'react-test-renderer';
import App from '../App';
import Button from '../components/Button/Button';
import ButtonPanel from '../components/ButtonPanel/ButtonPanel';
import Display from '../components/Display/Display';

it('renders App correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Button correctly', () => {
  const tree = renderer
    .create(<Button />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ButtonPanel correctly', () => {
  const tree = renderer
    .create(<ButtonPanel />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Display correctly', () => {
  const tree = renderer
    .create(<Display />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
