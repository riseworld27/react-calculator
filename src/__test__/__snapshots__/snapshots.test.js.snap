// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div
  className="container"
>
  <div
    className="buttonPanel"
  >
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f49040",
          "color": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      /
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f49040",
          "color": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      x
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f49040",
          "color": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      -
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f49040",
          "color": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      +
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "50%",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfdfdf",
          "color": "black",
          "width": "25%",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f49040",
          "color": "white",
          "width": "25%",
        }
      }
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`renders Button correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#f49040",
      "color": "white",
      "width": "25%",
    }
  }
  type="button"
>
  Empty
</button>
`;

exports[`renders ButtonPanel correctly 1`] = `
<div
  className="buttonPanel"
>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    AC
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    +/-
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    %
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#f49040",
        "color": "white",
        "width": "25%",
      }
    }
    type="button"
  >
    /
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    7
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    8
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    9
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#f49040",
        "color": "white",
        "width": "25%",
      }
    }
    type="button"
  >
    x
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    4
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    5
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    6
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#f49040",
        "color": "white",
        "width": "25%",
      }
    }
    type="button"
  >
    -
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    1
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    2
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    3
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#f49040",
        "color": "white",
        "width": "25%",
      }
    }
    type="button"
  >
    +
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "50%",
      }
    }
    type="button"
  >
    0
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "color": "black",
        "width": "25%",
      }
    }
    type="button"
  >
    .
  </button>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#f49040",
        "color": "white",
        "width": "25%",
      }
    }
    type="button"
  >
    =
  </button>
</div>
`;

exports[`renders Display correctly 1`] = `
<div
  className="displayContainer"
>
  <p
    className="displayOutput"
  >
    0
  </p>
</div>
`;
